src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -1166,17 +1169,18 @@ size_t ObjectSynchronizer::in_use_list_ceiling() {
 
 void ObjectSynchronizer::dec_in_use_list_ceiling() {
   Atomic::add(&_in_use_list_ceiling, (jint)-AvgMonitorsPerThreadEstimate);
-#ifdef ASSERT
-  size_t l_in_use_list_ceiling = in_use_list_ceiling();
-#endif
-  assert(l_in_use_list_ceiling > 0, "in_use_list_ceiling=" SIZE_FORMAT
-         ": must be > 0", l_in_use_list_ceiling);
 }
 
 void ObjectSynchronizer::inc_in_use_list_ceiling() {
   Atomic::add(&_in_use_list_ceiling, (jint)AvgMonitorsPerThreadEstimate);
 }
 
+void ObjectSynchronizer::set_in_use_list_ceiling(size_t new_value) {
```

Comment:
new_value should just be declared as int or jint now and the cast and comments removed.

Reply:
I don't think so. For some reason @fisk used size_t for MonitorList
_count and _max fields so the ceiling that is passed in and returned
should be size_t. The only reason that the _in_use_list_ceiling is jint
is because the range is 0..max_jint. At one point I tried to change it
to size_t and ran into build issues, but I don't remember the details.